analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.cs

```diff
@@ -218,13 +219,51 @@ public void Branching_ConstraintTrackedSeparatelyInBranches()
     value = false;
 }
 Tag(""End"", value);";
-            var validator = SETestContext.CreateCS(code).Validator;
-            validator.ValidateExitReachCount(2);    // Once with True constraint, once with False constraint on "value"
+            var validator = SETestContext.CreateCS(code, new PreserveTestCheck("value")).Validator;
+            validator.ValidateExitReachCount(2); // Once with True constraint, once with False constraint on "value"
             validator.TagValues("End").Should().HaveCount(2)
                 .And.ContainSingle(x => x.HasConstraint(BoolConstraint.True))
                 .And.ContainSingle(x => x.HasConstraint(BoolConstraint.False));
         }
 
+        [TestMethod]
+        public void Branching_FieldSymbolsAreNotCleaned()
```

Comment:
This and the next new UT has nothing to do with branching logic. They should be in `RoslynSymbolicExzecutionTest.cs` instead, because they care about state persistence during execution.

They should be also renamed accordingly (`Branching_` => `Execute_`)